Sources/Sentry/SentryOutOfMemoryScopeObserver.m

```diff
@@ -0,0 +1,155 @@
+#import "SentryOutOfMemoryScopeObserver.h"
+#import <SentryBreadcrumb.h>
+#import <SentryFileManager.h>
+#import <SentryLog.h>
+
+@interface
+SentryOutOfMemoryScopeObserver ()
```

Comment:
l: This is really bothering me.
Can we rename this for `BreadcrumbScopeObserver` or something else that says what this class does?
There is nothing in this class that is related to OOM, beside the fact that OOMTracker is the only thing using this right now.
In the future we will need this for App hang crashes, and probably other type of crashes.

Reply:
Shouldn't we just rename it then? At the moment it's only used on OOM, and it's instantiated within `SentryOutOfMemoryTrackingIntegration`, so right now it really is the scope observer for OOM. But I have no strong feelings about it either.

Reply:
IMO this is a breadcrumbs observer used by OOM, because what this really observe is breadcrumbs being created. 
But this is just my OCD.

Overall I'm happy with the PR.

Don't worry about the other comment regarding "every event". Maybe we'll come back to that in a future feature.

Reply:
> future feature.

Try saying that 10 times in a row :)

I think the way I named it now makes the most sense to me because I don't see what you're planning to use the same observer for, in the future. That whole "every event" discussion. So maybe it would be good to explain your thinking after all so we're on the same page?